cron: Added comment for completed RFCs

Bullrich opened the pull request:
Added the ability to check on open PRs if they match with a completed referenda.

If it matches, it will comment on it for the system to close it using the latest block from the timeline

## Summary

- **New Features**
	- Enhanced GitHub action notifications for voting availability and closure of referenda.
	- Added functionality to automatically close PRs for completed referenda.
- **Documentation**
	- Updated README with a short summary of new GitHub action notifications.
- **Refactor**
	- Improved handling of ongoing and completed referenda in the application logic.
- **New Files**
	- Introduced `SubsquareApi` class for fetching referenda data.

mordamax commented on src/cron.ts:
```diff
@@ -127,13 +151,27 @@ export const cron = async (startDate: Date, owner: string, repo: string, octokit
+      const closedMatch = completed.find(({ hash }) => hash === tx.method.hash.toHex() || hash === tx.method.toHex());
+      if (closedMatch) {
+        logger.info(`Found completed referenda for PR #${pr}`);
+        const msg = `/rfc process ${closedMatch.executedHash}`;
```
so it will request bot to check and merge?
may be better to start from suggesting this command to `copy` and paste by owner? for at least first few weeks, and then if it all work as expected - make it automatic

rzadp commented on src/subsquare.ts:
```diff
@@ -0,0 +1,42 @@
+export class SubsquareApi {
+  private readonly url: string;
+  constructor() {
+    this.url = "https://collectives.subsquare.io";
```
It would be consistent to add this to the constants:

https://github.com/paritytech/rfc-action/blob/4d81e1f325181fee9ab8789bbc5394a4268cc7ef/src/constants.ts#L1

Bullrich replied:
Good idea! Done!

rzadp commented on src/cron.test.ts:
```diff
@@ -2,7 +2,15 @@ import { getAllRFCRemarks } from "./cron";
+    expect(completed).toHaveLength(0);
+  }, 90_000);
+
+  test("Should return completed projects in the past", async () => {
```
This test is using external api and it's gonna keep growing (with every new referenda on the chain).

So with every test run, we will keep hitting subsquare with requests about all referenda.

Would it make sense to add an optional `endDate` parameter, and use it here so we keep requesting a small, constant amount of referenda from subsquare?

Bullrich replied:
Actually, you gave me a great idea!

The approval object is something along the lines of `[ 1603062, null, null ]`.

That first number is the block time.

And it is the same time as the one I'm getting from the subsquare api. I can check that before hand!

Bullrich replied:
It works!

I'll be moving the subsquare api to a different method so we can properly test it with the tests

rzadp commented on src/cron.ts:
```diff
@@ -57,11 +63,17 @@ export const getAllRFCRemarks = async (startDate: Date): Promise<{ url: string;
       logger.info(`Fetching elements ${index + 1}/${query}`);
 
-      const refQuery = (await api.query.fellowshipReferenda.referendumInfoFor(index)).toJSON() as { ongoing?: OnGoing };
+      const refQuery = (await api.query.fellowshipReferenda.referendumInfoFor(index)).toJSON() as {
```
It's a little confusing to call it a query, at this point the query was already executed and we have the result info.

How about calling it `refInfo`?

Bullrich commented on src/cron.ts:
```diff
@@ -89,12 +101,29 @@ export const getAllRFCRemarks = async (startDate: Date): Promise<{ url: string;
         }
       } else {
         logger.debug(`Reference query is not ongoing: ${JSON.stringify(refQuery)}`);
```
Great idea! I merged them together and added a debug log for an unhandled type (which will be rejected rfcs eventually)

Bullrich commented:
Hi @rzadp!

Thank you for your thorough review! You had a lot of great points!

I have addressed all of them. I also rewrite a bit the subsquare logic which should now make it more straightforward (and fixed the issue with the tests using an api)
